### PR TITLE
[GAL-2037] Move collection to other gallery

### PR DESCRIPTION
**Pull request by jakzaizzat**
**Changes**

- Add Move collection to other gallery
- Add markdown support to toast

**Demo**

**Move collection flow**

https://user-images.githubusercontent.com/4480258/215779642-a50bf98f-ff78-4d49-bbc5-048da73b1160.mp4

**Move collection flow - edge cases**

1. Add token
2. Move collection to different gallery
3. Remove token again

Expected: Its not **unsaved changes**

https://user-images.githubusercontent.com/4480258/215780080-115fdb95-1910-46b9-b1ba-1af43bfe125e.mp4

**Modal with a lot of gallery**

https://user-images.githubusercontent.com/4480258/215782564-0c4bbc11-4ebc-47aa-a371-b6610bcc4d5e.mp4

### CODE REVIEW

File: src/contexts/toast/Toast.tsx [hunk 1]
```diff
@@ -98,7 +99,9 @@ function Toast({ message, onClose, variant }: Props) {
             <AlertIcon />
           </StyledAlertIcon>
         )}
-        <BaseM>{message}</BaseM>
+        <BaseM>
+          <Markdown text={message} />
+        </BaseM>
```
Comment (jakzaizzat) [inline src/contexts/toast/Toast.tsx:104]: Added markdown support. Been thinking to support `NodeElement` but it seems lot of refactor needed

Comment (Robinnnnn) [reply src/contexts/toast/Toast.tsx:104]: nice one!!

File: src/components/GalleryEditor/CollectionSidebar/MoveCollectionModal.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  ]);
+
+  const handleOnSelect = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    const selectedGalleryId = event.target.value;
+    setSelectedGalleryId(selectedGalleryId);
+  }, []);
+
+  const isButtonDisabled = useMemo(() => {
+    return !selectedGalleryId;
+  }, [selectedGalleryId]);
```
Comment (tbezman) [inline src/components/GalleryEditor/CollectionSidebar/MoveCollectionModal.tsx:114]: nit: no need for useMemo here

File: src/components/GalleryEditor/GalleryEditorContext.tsx [hunk 3]
```diff
@@ -436,6 +437,19 @@ export function GalleryEditorProvider({
...
+      const nextInitialCollections = { ...initialCollections };
+      delete nextInitialCollections[collectionId];
+      setInitialCollections(nextInitialCollections);
+
+      const nextCollections = { ...collections };
+      delete nextCollections[collectionId];
+      setCollections(nextCollections);
+    },
+    [collections, initialCollections]
+  );
```
Comment (tbezman) [inline src/components/GalleryEditor/GalleryEditorContext.tsx:451]: ❤️

**Review by tbezman — APPROVED** [review]
@Robinnnnn these strange edge cases around moving tokens into a collection you're about to move is more gunpowder to make this apart of the `saveGalleryMutation`. We should just have another field on the input type, `moveCollections` or something like that.

**Pull request merged**